### DISCUSSION TITLE
Upsert with set behaviour

### DISCUSSION THREAD

**Discussion by ChrisJohnson-83** [prisma/prisma#2998] [Q&A] [answered]
I am wondering if there is a way to use upsert in a way which would also allow records to be removed? I have a basic CRUD component which lets me add, edit, re-order and remove items.

![image](https://user-images.githubusercontent.com/8020291/87228631-cd050d00-c3a2-11ea-8cb3-4f3f7240d4d2.png)

I am using the `useFieldArray` hook from react-hook-form and then parsing the data via GraphQL to Primsa. Everything works with the exception of delete, I don't know how I can persist this in the database?

I guess I am looking for something like the `set` function but with the ability to create records rather than `connect` them. Or, another approach could be with a transaction I guess. Although I like the idea of being able to do this in a single transaction.

Does anyone know if this is possible?

```
export const updatePost = ({ id, input }) => {
  return db.post.update({
    data: {
      title: "My First Post",
      body: "Some body text..."
      comments: {
        upsert: {
        "create": {
          "message": "Edited message 1",
          "position": 1 
        },
        "update": { 
          "message": "Edited message 1",
          "position": 1
        },
        "where": {
          "id": 5
        }
      },
      {
        "create": { 
          "message": "Edited message 2",
          "position": 2 
        },
        "update": { 
          "message": "Edited message 2",
          "position": 2 
        },
        "where": {
          "id": 6
        }
      }
      },
    },
    where: { "id": 1 },
  })
}
```

**Comment by ryands17** [comment]
Hey @ChrisJohnson-83 :wave: 
I didn't quite get your question. Do you want create apart from upsert, or you want a simple delete?

**Comment by ChrisJohnson-83** [comment]
Hi @ryands17. My understanding is that `set` disconnects all existing items linked to a record and then connects one or more new records. I was just wondering if it was possible to achieve the same behaviour but using `create` rather than `connect`.

For now I have just done it using two transactions, the first is a `deleteMany` to remove all posts currently connected to a record and the second is a nested `create` to add the new records.

**Reply by ryands17** [reply, marked as answer]
Ohh yes, currently `create` in `set` isn't possible, as you have mentioned and your workaround seems to be the best option for the time being. It would be amazing if you could create a feature request for this explaining the use case that could be added to `set` [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
